internal/pkg/cli/pipeline_init.go

```diff
@@ -511,6 +524,15 @@ For more information, please refer to: https://git.io/JfDFD.`,
 }
 
 func (o *initPipelineOpts) getEnvs() ([]*config.Environment, error) {
+	// If app name is passed in via flag, validate here.
+	if o.appName == "" {
+		return nil, errValueEmpty
+	}
+	if o.appName != "" {
+		if _, err := o.store.GetApplication(o.appName); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
I still feel this validation should be in the `Validate()` since it is validating the flag value. However, the root is because `newInitPipelineOpts` is too messy. We should probably do some refactoring work to make it consistent with the other commands. Don't want to be a blocker just want to bring it up for visibility.

(It would be really nice if we can stop adding to this file after this PR and before the refactoring otherwise the tech debt continues to grow)